src/test/scala/org/infinispan/spark/test/InfinispanServer.scala

```diff
@@ -274,17 +275,24 @@ object SingleNode {
 
    def getServerPort = server.getHotRodPort
 
-   def getOrCreateCache(name: String) = server.obtainNonClusteredCache(name)
+   def getOrCreateCache(name: String, extraConfigs:Option[List[String]]) = {
+      server.addLocalCache(name, extraConfigs)
+      server.obtainRemoteCache(name)
+   }
 }
 
 object Cluster {
-   private val cluster: Cluster = new Cluster(3, getClass.getResource("/infinispan-server/").getPath)
+   val NumberOfServers = 3
+   val ServerPath = getClass.getResource("/infinispan-server/").getPath
+   val StartTimeout = 60 seconds
+
+   private val cluster: Cluster = new Cluster(NumberOfServers, ServerPath)
 
-   def start() = cluster.startAndWait(60 seconds)
+   def start() = cluster.startAndWait(StartTimeout)
 
    def shutDown() = cluster.shutDown()
 
-   def getOrCreateCache(name: String, cacheType: CacheType.Value) = cluster.obtainCache(name, cacheType)
+   def getOrCreateCache(name: String, cacheType: CacheType.Value, extraCacheConfigs: Option[List[String]]) = cluster.obtainCache(name, cacheType, extraCacheConfigs)
```

Comment:
Would it make sense to have a default value of `extraCacheConfigs = None` in the parameter list? Since the parameters seem optional, seems like a good case for a default value :)